### ISSUE TITLE
I need the GUI to generate the address of the wallet path-related file or registry, not the wallet file.

### ISSUE THREAD

**Issue by gcoulu** [gridcoin-community/Gridcoin-Research#2736]
**Try**
1.I uninstalled the GUI
2.Deleted part of the registry I found

**Question**
I cannot find the files in the disk other than the wallet file created by the GUI, and cannot modify or reset the wallet.

**Gridcoin version**
gridcoin-5.4.5-win64-setup

**Machine specs**
- OS:Windows 10
- CPU:Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz
- RAM:96GB(32G*3) 2133MHz DIMM
- Disk size:2TB
- Disk Type (HD/SSD):H730 array card

**information**
When creating a wallet via Gridcoin (64-bit)，the set wallet path contains some characters that may not be recognized correctly, resulting in garbled characters and making the wallet unable to be opened normally.Since uninstalling the program GUI does not clear the wallet's path information (note: this does not refer to the wallet's files), it cannot be re-created or reset through the GUI, so **I need the GUI to generate the address of the wallet path-related files**.

![image](https://github.com/gridcoin-community/Gridcoin-Research/assets/63546502/62d6d60f-5e12-4624-9115-a76db390a8df)

**Comment by jamescowens** [comment]
This looks like a locality issue with filename paths? Did you choose other than a default location to store the data files when you installed the wallet?

**Comment by gcoulu** [comment]
> This looks like a locality issue with filename paths? Did you choose other than a default location to store the data files when you installed the wallet?

Yes, I created the wallet with a custom path that contains UTF-8. I think this is caused by garbled characters.And **how to reset index files, I don't know where they are.**

**Comment by jamescowens** [comment]
I will lookup the registry entries as soon as I can get to my workstation and provide you a procedure to delete them.

**Comment by gcoulu** [comment]
I accidentally clicked the wrong button，I am curious about the solutions to such problems encountered in Linux and Windows

**Comment by jamescowens** [comment]
I think we need to provide a way to erase the registry entries during the install. The installer specifically does NOT disturb the data directory or the registry entries pointing to it done by a prior install unless they are messed up. This is to enable easy in place upgrades, and protects people from inadvertently deleting their data directory and destroying their coins.

**Comment by gcoulu** [comment]
> I think we need to provide a way to erase the registry entries during the install. The installer specifically does NOT disturb the data directory or the registry entries pointing to it done by a prior install unless they are messed up. This is to enable easy in place upgrades, and protects people from inadvertently deleting their data directory and destroying their coins.

Agreed, this is particularly useful when needed. I think it can be independently developed as a program or script, published separately on the official website or Git platform, and provided for download by those in need. This can avoid the occurrence of most misoperations.

**Comment by gcoulu** [comment]
I want to add a language selection option to the initialization interface of the wallet (currently, the initialization interface cannot change the language through GUI), as people using this software come from all over the world. At the same time, add a warning in red or bold to the custom wallet path, stating "Do not use directories containing languages or characters other than English"

**Comment by jamescowens** [comment]
I think the options can be included in the NSIS install script. I have marked this as a feature request.

**Comment by gcoulu** [comment]
> I think the options can be included in the NSIS install script. I have marked this as a feature request.

Looking forward to the process of manually deleting registry entries，Looking forward to the emergence of new install script！

**Comment by jamescowens** [comment]
Investigated and addressed in PR #2863.

**Root cause:** `qstringToBoostPath()` in `guiutil.cpp` converts `QString` → `std::string` via `toStdString()` (UTF-8), then passes the narrow string to `boost::filesystem::path`. On Windows, Boost interprets narrow strings using the system code page (e.g. CP-1252 for Western European locales, GBK for Chinese), not UTF-8. Any characters outside the system code page get garbled — which is what happened with the Chinese-character custom data directory path reported here.

The reverse conversion in `boostPathToQString()` has the same problem: `path.string()` produces code page bytes, but `QString::fromStdString()` expects UTF-8.

Once the path is garbled and stored in the Windows registry via QSettings, the wallet can't find its data directory on subsequent launches. The user then has no way to recover without manually adding `-choosedatadir` to the shortcut target or editing the registry — neither of which is a reasonable expectation for end users.

**Fix (three parts):**
1. **Encoding:** On Windows, use `toStdWString()`/`fromStdWString()` and `path.wstring()` to stay in UTF-16 (the native Windows path encoding). Linux/macOS are unaffected since UTF-8 narrow strings work correctly there.
2. **Inline occurrences:** Replaced four additional inline `QString::fromStdString(path.string())` patterns in `bitcoingui.cpp`, `optionsmodel.cpp`, and `researchermodel.cpp` with the corrected `boostPathToQString()`.
3. **Auto-recovery:** `Intro::showIfNeeded()` now checks `fs::exists()` on the stored data directory path before the early return. If the path doesn't exist (whether from encoding corruption, manual relocation, or deletion), the directory chooser dialog appears automatically instead of failing silently. This eliminates the need for the `-choosedatadir` workaround.

Regarding the request for registry key deletion — the auto-recovery in part 3 effectively addresses this. When the stored path is invalid, the chooser dialog lets the user reselect, and the new valid path overwrites the old QSettings entry. An explicit registry deletion mechanism in the installer is no longer necessary for this failure mode, though it could still be considered as a general maintenance feature in the future.
